### ISSUE TITLE
CoreDNS externalCoreFile regression on v1.16

### ISSUE THREAD

**Issue by rochacon** [kubernetes/kops#8642]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

```
$  kops version
Version 1.16.0 (git-4b0e62b82)
```

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

v1.16.7

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

```kops create -f cluster.yaml && kops update cluster ...```

**5. What happened after the commands executed?**

`kubeDNS.externalCoreFile` failed to render with incorrect YAML formatting.

Error message:

```
error building tasks: error remapping manifest addons/coredns.addons.k8s.io/k8s-1.6.yaml: error parsing yaml: error converting YAML to JSON: yaml: line 12: could not find expected ':'
```

**6. What did you expect to happen?**

#7979 fix the issue, apparently it was merged into 1.15 but not 1.16 release branch. 1.17 seems OK.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

Relevant part:

```
---
apiVersion: kops.k8s.io/v1alpha2
kind: Cluster
metadata:
  name: k.aws.rochacon.me
spec:
  # ...
  kubeDNS:
    provider: CoreDNS
    coreDNSImage: 602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.6.7
      externalCoreFile: |
        amazonaws.com:53 {
          cache 30
          errors
          loadbalance
          log . {
            class denial error
          }
          prometheus :9153
          forward . 169.254.169.253
        }
        .:53 {
          autopath @kubernetes
          cache 90
          errors
          health {
            lameduck 5s
          }
          loadbalance
          log . {
            class denial error
          }
          loop
          kubernetes cluster.local. in-addr.arpa ip6.arpa {
            fallthrough in-addr.arpa ip6.arpa
            pods verified
            upstream 169.254.169.253
          }
          prometheus :9153
          forward . 169.254.169.253
          reload
        }
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

```
$ kops update cluster ...
# ... 
error building tasks: error remapping manifest addons/coredns.addons.k8s.io/k8s-1.6.yaml: error parsing yaml: error converting YAML to JSON: yaml: line 12: could not find expected ':'
```

**9. Anything else do we need to know?**

This seems like a backporting issue, it might affect other functionality too.

**Comment by rifelpet** [comment]
Thanks for the report! I just opened a cherry-pick for 1.16, so it will be included in Kops 1.16.1.

**Comment by rochacon** [comment]
Thanks @rifelpet !
